packages/zend-db/library/Zend/Db/Statement/Mysqli.php

```diff
@@ -67,7 +67,13 @@ public function _prepare($sql)
     {
         $mysqli = $this->_adapter->getConnection();
 
-        $this->_stmt = $mysqli->prepare($sql);
+        // https://www.php.net/manual/en/class.mysqli-sql-exception.php
+        // https://php.watch/versions/8.1/mysqli-error-mode
+        try {
+            $this->_stmt = $mysqli->prepare($sql);
+        } catch (\mysqli_sql_exception $e) {
+            throw \Zend_Db_Statement_Mysqli_Exception::fromMysqliException($e);
```

Comment:
There is no need to use namespace qualifier as zf1 is not namespaced.
```suggestion
        } catch (mysqli_sql_exception $e) {
            throw Zend_Db_Statement_Mysqli_Exception::fromMysqliException($e);
```